packages/instantsearch.js/src/lib/server.ts

```diff
@@ -37,17 +57,27 @@ export function waitForResults(search: InstantSearch): Promise<void> {
 /**
  * Walks the InstantSearch root index to construct the initial results.
  */
-export function getInitialResults(rootIndex: IndexWidget): InitialResults {
+export function getInitialResults(
+  rootIndex: IndexWidget,
+  /**
+   * Search parameters sent to the search client, usually
+   * returned by `waitForResults()`.
+   */
+  requestParamsList?: RequestParams[]
+): InitialResults {
   const initialResults: InitialResults = {};
 
   walkIndex(rootIndex, (widget) => {
     const searchResults = widget.getResults();
     if (searchResults) {
-      initialResults[widget.getIndexId()] = {
+      const indexId = widget.getIndexId();
+      const requestParams = requestParamsList?.shift();
+      initialResults[indexId] = {
         // We convert the Helper state to a plain object to pass parsable data
         // structures from server to client.
         state: { ...searchResults._state },
         results: searchResults._rawResults,
+        ...(requestParams && { requestParams }),
```

Comment:
```suggestion
       requestParams,
```

What's the difference with this? undefined if it's not available?

Reply:
Yes, if it's `undefined` it will throw because Next won't serialize it as JSON.

The 2 options are either omitting the property or returning `null`. I chose the first option in this case.